### PR TITLE
Support JSON.ARRINDEX #486

### PR DESCRIPTION
**Pull request by iamskp11**
Implements #486.

### CODE REVIEW

File: internal/eval/store_eval.go [hunk 1]
```diff
@@ -6781,3 +6782,134 @@ func evalCommandDocs(args []string) *EvalResponse {
...
+// It handles negative indices and ensures they are within the array bounds. 
+func adjustIndices(start, stop, length int) (adjustedStart, adjustedStop int) { 
+	if length == 0 {
+		return -1, -1 
+	}
+	if start < 0 {
+		start += length 
+	}
+
+	// 0 or -1 means the last element is included.
```
Comment (iamskp11) [inline internal/eval/store_eval.go:6895]: ToDo: Modify this to correct comment.

File: internal/eval/store_eval.go [hunk 2]
```diff
@@ -6781,3 +6782,134 @@ func evalCommandDocs(args []string) *EvalResponse {
...
+		var err error
+		stop, err = strconv.Atoi(args[4])
+		if err != nil {
+			return diceerrors.NewErrWithMessage("Couldn't parse as integer")
+		}
+	}
+
+	obj := store.Get(key)
+	if obj == nil {
+		return diceerrors.NewErrWithMessage("object with key does not exist")
```
Comment (iamskp11) [inline internal/eval/store_eval.go:6831]: ToDo: Match with dicedb output

File: internal/eval/store_eval.go [hunk 3]
```diff
@@ -6781,3 +6782,134 @@ func evalCommandDocs(args []string) *EvalResponse {
...
+	// Check if the value stored is JSON type
+	_, err = sonic.Marshal(jsonData)
+	if err != nil {
+		return diceerrors.NewErrWithMessage("existing key has wrong Dice type")
+	}
+
+	// Check if the path specified is valid
+	expr, err := jp.ParseString(path)
+	if err != nil {
+		return diceerrors.NewErrWithMessage("invalid JSONPath")
```
Comment (iamskp11) [inline internal/eval/store_eval.go:6845]: ToDo: Match with dicedb output

**Review by apoorvyadav1111 — APPROVED** [review]

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Could you please also add documentation for this new command?

Thanks,
Apoorv

**Comment by apoorvyadav1111** [comment]
Hi @iamskp11, I reran the CI for the PR. Also, I would like to request to change the permutation function in other two files as well.

Thanks

**Comment by apoorvyadav1111** [comment]
Hi @iamskp11 , thanks for these changes. Could you please rebase the PR? I can merge once the CI checks are passed.

**Review by apoorvyadav1111 — APPROVED** [review]

**Comment by iamskp11** [comment]
@apoorvyadav1111  Pls do not merge. Getting some issues related to HTTP tests since cmd_meta.go is deleted in latest master. Will debug and fix today later.

**Comment by JyotinderSingh** [comment]
> @apoorvyadav1111 Pls do not merge. Getting some issues related to HTTP tests since cmd_meta.go is deleted in latest master. Will debug and fix today later.

Hi @iamskp11, did you get a chance to address the issues?

**Comment by iamskp11** [comment]
> > @apoorvyadav1111 Pls do not merge. Getting some issues related to HTTP tests since cmd_meta.go is deleted in latest master. Will debug and fix today later.
> 
> Hi @iamskp11, did you get a chance to address the issues?

Hi @JyotinderSingh , thanks for reminding.
Sorry, wasn't able to prioritise it last week. This is fixed, and tests are passing.
But, it failed Github check here, "context cancelled" issue. Can you re-run the tests?

**Comment by apoorvyadav1111** [comment]
Hi @iamskp11 , Is this PR ready for review and merge? Please confirm

**Comment by iamskp11** [comment]
> Hi @iamskp11 , Is this PR ready for review and merge? Please confirm

Yes @apoorvyadav1111 , this is ready for review, merge. Have added latest changes.

**Pull request merged**